lib/camunda-cloud/ElementTemplatesReplaceProvider.js

```diff
@@ -0,0 +1,105 @@
+import {
+  getBusinessObject,
+  isAny
+} from 'bpmn-js/lib/util/ModelUtil';
+
+/**
+ * A replace menu provider that allows to replace elements with
+ * element templates.
+ */
+function ElementTemplatesReplaceProvider(popupMenu, translate, elementTemplates) {
+
+  this._popupMenu = popupMenu;
+  this._translate = translate;
+  this._elementTemplates = elementTemplates;
+
+  this.register();
+}
+
+ElementTemplatesReplaceProvider.$inject = [
+  'popupMenu',
+  'translate',
+  'elementTemplates'
+];
+
+
+/**
+ * Register replace menu provider in the popup menu
+ */
+ElementTemplatesReplaceProvider.prototype.register = function() {
+  this._popupMenu.registerProvider('bpmn-replace', this);
+};
+
+/**
+ * Get all element templates that can be used to replace the given element.
+ *
+ * @param {djs.model.Base} element
+ *
+ * @return {Array<Object>} a list of element templates as menu entries
+ */
+ElementTemplatesReplaceProvider.prototype.getEntries = function(element) {
+
+  const templates = this._getMatchingTemplates(element);
+  return templates.map(template => {
```

Comment:
We'd need to ensure that we can replace from a "service task with a template applied" to a plain service task, too. 

Let's test cover that case, and consider if we need to improve our core replace infrastructure (bpmn-js) to properly support that.